### DISCUSSION TITLE
Compute how large a D2C message will be

### DISCUSSION THREAD

**Discussion by drwill-ms** [Azure/azure-iot-sdk-csharp#2106] [General]
**Is your feature request related to a problem? Please describe.**
Give the user feedback about how large a message will be so it can be taken into account for message batching.

**Describe the solution you'd like**
It needs to take into account all the payload that the service uses to determine message size, some of which is not available to the user.

**Additional context**
User (@Marusyk) asking for this feature here: <https://github.com/Azure/azure-sdk-for-net/issues/16969>.

**Comment by barnstee** [comment]
Using the full payload available for a large IoT Hub message right now is a guessing game as the system property size is not exposed but subtracted from the maximum available 256K message size. Please implement this feature request.
